spinalcordtoolbox/metadata.py

```diff
@@ -210,31 +211,26 @@ def get_file_label(path_label='', id_label=0, output='file'):
 
     raise RuntimeError("Label ID {} not found in {}".format(id_label, fname_label))
 
-def get_indiv_label_names(directory):
+def get_indiv_label_info(directory):
     """
-    Get all individual label names in a folder
+    Get all individual label info (id, name, filename) in a folder
     :param directory: folder containing info_label.txt and the files
-    :return: the labels (strings)
+    :return: dictionary containing "id_label" the label IDs (int),
+                                    "name_label" the labels (string),
+                                    "file_label" the label filename (string)
     """
 
     file_info_label = 'info_label.txt'
     il = InfoLabel()
     fname_label = os.path.join(directory, file_info_label)
     il.load(fname_label)
 
-    return tuple([_name for (_id, _name, _file) in il._indiv_labels])
+    id_lst = list(map(itemgetter(0), il._indiv_labels))
+    name_lst = list(map(itemgetter(1), il._indiv_labels))
+    filename_lst = list(map(itemgetter(2), il._indiv_labels))
 
-def get_indiv_label_ids(directory):
-    """
-    Get all individual label IDs in a folder
-    :param directory: folder containing info_label.txt and the files
-    :return: the ids (int)
-    """
-
-    file_info_label = 'info_label.txt'
-    il = InfoLabel()
-    fname_label = os.path.join(directory, file_info_label)
-    il.load(fname_label)
-
-    return tuple([id_ for (_id, _name, _file) in il._indiv_labels])
+    return {'id_label': tuple(id_lst),
```

Comment:
i would use the same field names as in the info_label file, i.e.: 'id', 'name' and 'file' (although strictly speaking id should be capitalized...)

Reply:
Done :+1: